redwood-compose/src/commonTest/kotlin/app/cash/redwood/compose/ChangeListenerTest.kt

```diff
@@ -91,13 +93,19 @@ abstract class AbstractChangeListenerTest {
     c.setContent {
       Button(text, onClick = null)
     }
-    assertThat(c.awaitSnapshot()).containsExactly("modifier Modifier", "text hi", "onClick false", "onEndChanges")
+    assertThat(c.awaitSnapshot()).containsExactlyInAnyOrder(
```

Comment:
This does not seem like a desirable effect. This test will now pass if `onEndChanges` is called before some property or the modifier is set. Why is the behavior changing between regular Redwood and protocol Redwood? That also seems like an indicator something is wrong.

Reply:
I figured this out later, and proposed some changes to avoid needing to relax this